bin/lib/common.sh

```diff
@@ -0,0 +1,318 @@
+  # shellcheck source=/dev/null
+  source "${env_file}"
+  set +o allexport
+}
+
+source_osmt_env_file() {
+  local env_file="${1}"
+
+ # gracefully bypass sourcing env file if these 4 OAUTH values are provided
+  if [[ \
+      -n "${OAUTH_ISSUER}" && \
+      -n "${OAUTH_CLIENTID}" &&\
+      -n "${OAUTH_CLIENTSECRET}" && \
+      -n "${OAUTH_AUDIENCE}" \
+    ]]; then
+      echo_info "OAUTH values are provided by environment variables. Not sourcing ${env_file} env file."
+      return 0
+    fi
+
+  source_env_file "${env_file}"
+}
+
+start_osmt_docker_stack() {
+  local stack_name="${1}"
+  local -i rc
+  echo
+  echo_info "Starting OSMT ${stack_name} Docker stack. You can stop it with $(basename "${0}") -e"
+  cd "${PROJECT_DIR}/docker" || return 1
+  # Docker stack should receive the service port variables sourced from the shell environment
+  docker-compose --file dev-stack.yml --project-name "${stack_name}" up --detach
+  rc=$?
+  if [[ $rc -ne 0 ]]; then
+    echo_err "Starting OSMT ${stack_name} Docker stack failed. Exiting..."
+    return 1
+  fi
+}
+
+stop_osmt_docker_stack() {
+  local stack_name="${1}"
+  local -i rc
+  echo
+  echo_info "Stopping OSMT ${stack_name} Docker stack"
+  cd "${PROJECT_DIR}/docker" || return 1
+  docker-compose --file dev-stack.yml --project-name "${stack_name}" down
+  rc=$?
+  if [[ $rc -ne 0 ]]; then
+    echo_err "Stopping OSMT ${stack_name} Docker stack failed. Exiting..."
+    return 1
+  fi
+}
+
+remove_osmt_docker_artifacts_for_stack() {
+  local stack_name="${1}"
+
+  stop_osmt_docker_stack "${stack_name}"
+
+  echo_info "Removing Docker containers for ${stack_name}..."
+  docker ps -aq --filter=name="${stack_name}" | xargs docker rm
+
+  echo_info "Removing Docker images for ${stack_name}..."
+  docker images -q --filter=reference="${stack_name}" | xargs docker rmi
+
+  echo_info "Removing Docker volumes for ${stack_name}..."
+  docker volume ls -q -f name="${stack_name}" | xargs docker volume rm -f {} &> /dev/null
+
+  echo_info "Removing Docker networks for ${stack_name}..."
+  docker network ls -q -f name="${stack_name}" | xargs docker network rm -f {} &> /dev/null
+}
+
+echo_info() {
+  echo -e "INFO:  $*"
+}
+
+echo_warn() {
+  echo -e "WARN:  $*"
+}
+
+echo_err() {
+  echo -e "ERROR: $*" 1>&2;
+}
+
+echo_debug() {
+  [[ "${DEBUG}" -ne 0 ]] && echo -e "DEBUG: $*"
+}
+
+
+#start_osmt_spring_app_via_maven() {
+#  echo
+#}
+#
+#start_osmt_spring_app_via_java() {
+#  echo
+#}
+
+_validate_env_file() {
+  local env_file="${1}"
+
+  echo
+  if [[ ! -f "${env_file}" ]]; then
+    echo_err "Can not access ${env_file}. You can initialize OSMT local environment files by running $(basename "${0}") -i"
+    return 1
+  fi
+
+  echo_info "Checking ${env_file} for invalid default values (xxxxxx)"
+  if grep -q "xxxxxx" "${env_file}"; then
+    echo_err "${env_file} should be updated with valid OAUTH2 values"
+    return 1
+  else
+    echo_info "No invalid default values in ${env_file}"
```

Comment:
I think we should not worry about manifest that there were not invalid default values provided here, only when  there are.